node/src/types/block/hot_block/state.rs

```diff
@@ -0,0 +1,244 @@
+use datasize::DataSize;
+use serde::Serialize;
+
+use super::MergeMismatchError;
+
+#[derive(Clone, Copy, Eq, PartialEq, Debug, DataSize)]
+pub(crate) enum StateChange {
+    Updated,
+    AlreadySet,
+}
+
+impl From<bool> for StateChange {
+    fn from(current_state: bool) -> Self {
+        if current_state {
+            StateChange::AlreadySet
+        } else {
+            StateChange::Updated
+        }
+    }
+}
+
+#[derive(Clone, Copy, Eq, PartialEq, Default, Serialize, Debug, DataSize)]
+pub(crate) struct State {
+    pub(super) is_immediate_switch_block_for_current_protocol_version: bool,
+    pub(super) is_stored: bool,
+    pub(super) has_been_sent_to_deploy_buffer: bool,
+    pub(super) has_updated_validator_matrix: bool,
+    pub(super) has_been_gossiped: bool,
+    pub(super) is_executed: bool,
+    pub(super) we_have_tried_to_sign: bool,
+    pub(super) has_been_sent_to_consensus_post_execution: bool,
+    pub(super) has_been_sent_to_accumulator_post_execution: bool,
+    pub(super) has_sufficient_finality: bool,
+    pub(super) is_marked_complete: bool,
```

Comment:
Function names can stay the same, but variable names could be shorter and more concise.
```suggestion
    pub(super) immediate_switch_block_for_current_protocol_version: bool,
    pub(super) stored: bool,
    pub(super) sent_to_deploy_buffer: bool,
    // Validator matrix updates can happen multiple times, so it's not a one-and-done thing like the rest. Are we sure we want this field here?
    pub(super) updated_validator_matrix: bool,
    pub(super) gossiped: bool,
    pub(super) executed: bool,
    pub(super) tried_to_sign: bool,
    pub(super) sent_to_consensus_post_execution: bool,
    pub(super) sent_to_accumulator_post_execution: bool,
    pub(super) has_sufficient_finality: bool,
    pub(super) marked_complete: bool,
```

Reply:
Will do the renames.  As for the comment about validator matrix update - the intention of the flag is to only notify the interested components once for this block, which I think is correct.

Reply:
Yes, the name threw me off but it's used correctly.

Reply:
Updated in [e70a836](https://github.com/casper-network/casper-node/pull/3549/commits/e70a836f70a16c5c9508184aa6ebafddfb6418a5).